assimilation_code/modules/assimilation/algorithm_info_mod.f90

```diff
@@ -128,107 +290,198 @@ subroutine probit_dist_info(kind, is_state, is_inflation, dist_type, &
 ! In the long run, may not have to have separate controls for each of the input possibilities
 ! However, for now these are things that need to be explored for science understanding
 
-if(is_inflation) then
+! Fill arrays with possible dist_type strings and corresponding ints
+possible_dist_types(1) = 'NORMAL_DISTRIBUTION'
+possible_dist_types(2) = 'BOUNDED_NORMAL_RH_DISTRIBUTION'
+possible_dist_types(3) = 'GAMMA_DISTRIBUTION'
+possible_dist_types(4) = 'BETA_DISTRIBUTION'
+possible_dist_types(5) = 'LOG_NORMAL_DISTRIBUTION'
+possible_dist_types(6) = 'UNIFORM_DISTRIBUTION'
+possible_dist_types(7) = 'PARTICLE_FILTER_DISTRIBUTION'
+
+possible_dist_type_ints(1) = 1
+possible_dist_type_ints(2) = 2
+possible_dist_type_ints(3) = 3
+possible_dist_type_ints(4) = 4
+possible_dist_type_ints(5) = 5
+possible_dist_type_ints(6) = 6
+possible_dist_type_ints(7) = 7
+
+!use default values if qcf_table_filename is not in namelist
+if (.not. qcf_table_listed) then
+   dist_type = BOUNDED_NORMAL_RH_DISTRIBUTION
+   bounded_below = .false.;    bounded_above = .false.
+   lower_bound   = missing_r8; upper_bound   = missing_r8
+   return
+endif
+
+!get actual name of QTY from integer index
+qty_name = get_name_for_quantity(kind)
+
+!find location of QTY in qcf_table_data structure
+QTY_loc = findloc(qcf_table_row_headers, qty_name)
+
+if (QTY_loc(1) == 0) then
+   !use default values if QTY is not in table
+   dist_type = BOUNDED_NORMAL_RH_DISTRIBUTION
+   bounded_below = .false.;    bounded_above = .false.
+   lower_bound   = missing_r8; upper_bound   = missing_r8
+
+   elseif(is_inflation) then
    ! Case for inflation transformation
-   if(kind == QTY_STATE_VARIABLE) then
-      dist_type = BOUNDED_NORMAL_RH_DISTRIBUTION
-      bounded_below = .false.;    bounded_above = .false.
-      lower_bound   = missing_r8; upper_bound   = missing_r8
-   elseif(kind == QTY_TRACER_CONCENTRATION) then
-      dist_type = BOUNDED_NORMAL_RH_DISTRIBUTION
-      bounded_below = .true.; bounded_above = .false.
-      lower_bound   = 0.0_r8; upper_bound = missing_r8
-   elseif(kind == QTY_TRACER_SOURCE) then
-      dist_type = BOUNDED_NORMAL_RH_DISTRIBUTION
-      bounded_below = .true.; bounded_above = .false.
-      lower_bound   = 0.0_r8; upper_bound   = missing_r8
-   else
-      write(*, *) 'Illegal kind in obs_error_info'
-      stop
-   endif
-elseif(is_state) then
+
+      ! Comparing the dist_type in string format to list of potential dist_types
+      dist_type_string = qcf_table_data(QTY_loc(1))%probit_inflation%dist_type
+      call to_upper(dist_type_string)
+      dist_type_loc = findloc(possible_dist_types, trim(dist_type_string))
+
+      if (dist_type_loc(1) == 0) then
+         write(errstring, *) 'Invalid dist_type: ', trim(dist_type_string)
+         call error_handler(E_ERR, 'probit_dist_info:', errstring, source)
+
+      else
+         dist_type = possible_dist_type_ints(dist_type_loc(1))
+      endif
```

Comment:
This dist_type string comparison only needs to be done once when you read the table.